pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs

```diff
@@ -235,77 +303,69 @@ private void TraceSouthWest(float[,] histogram, int bestXi, int bestYi, LinkedLi
                 var west = xi - 1 >= 0 ? histogram[xi - 1, yi] : -1;
                 var southwest = south != -1 && west != -1 ? histogram[xi - 1, yi - 1] : -1;
                 var max = Math.Max(south, Math.Max(west, southwest));
+                Assume.IsFalse(double.IsNaN(max));
                 if (max == -1)
                     break;
                 if (southwest == max || south == west)
                 {
-                    xi --;
-                    yi --;
-                    _minXi --;
-                    _minYi --;
+                    xi--;
+                    yi--;
+                    _minXi--;
+                    _minYi--;
                 }
                 else if (south == max)
                 {
-                    yi --;
-                    _minYi --;
+                    yi--;
+                    _minYi--;
                 }
                 else
                 {
-                    xi --;
-                    _minXi --; 
+                    xi--;
+                    _minXi--;
                 }
                 points.AddFirst(new Tuple<int, int>(xi, yi));
             }
         }
 
-        private void StampOutHistogram(float[,] histogram, float[,] stamp, double[] xArr, double[] yArr, CancellationToken token)
-        {
-            for (int p = 0; p < xArr.Length; p ++)
-            {
-                token.ThrowIfCancellationRequested();
-                int x = (int)xArr[p];
-                int y = (int) yArr[p];
-                Stamp(histogram,stamp,x,y);
-            }    
-        }
-
         private void Stamp(float[,] histogram, float[,] stamp, int x, int y)
         {
-            var stampLength0 = stamp.GetLength(0); // GetLength is more expensive than you'd think
-            var stampLength1 = stamp.GetLength(1);
-            var histogramLength0 = histogram.GetLength(0);
-            var histogramLength1 = histogram.GetLength(1);
-
-            for (int i = x - stampLength0/2; i <= x + stampLength0 / 2; i++)
+            for (int i = x - stamp.GetLength(0) / 2; i <= x + stamp.GetLength(0) / 2; i++)

```

Comment:
Does the "GetLength is more expensive than you'd think" comment still apply, or has more recent profiling disproven it?

Reply:
I haven't reprofiled it. But I can remove the comment if it's distracting.

Reply:
Oh right - uninitentional undoing of an optimization, thanks. Dang, I love your reviews.